op-chain-ops/crossdomain/migrate.go

```diff
@@ -82,8 +83,18 @@ func MigrateWithdrawal(withdrawal *LegacyWithdrawal, l1CrossDomainMessenger *com
 		return nil, fmt.Errorf("cannot abi encode relayMessage: %w", err)
 	}
 
+	// Compute the cost of the calldata
+	dataCost := uint64(0)
+	for _, b := range data {
+		if b == 0 {
+			dataCost += params.TxDataZeroGas
+		} else {
+			dataCost += params.TxDataNonZeroGasEIP2028
+		}
+	}
+
 	// Set the outer gas limit. This cannot be zero
-	gasLimit := uint64(len(data)*16 + 200_000)
+	gasLimit := dataCost + 200_000
```

Comment:
IMO we should add a cap of ~25m here.

Reply:
In the unlikely chance it goes over 25m, then it would create a deposit that cannot be relayed

Reply:
Not having that check gives an extra 5 million buffer. Right now we have a loose definition around withdrawals, should there be footguns?

Reply:
Ok i see why, will update